far/editor.cpp

```diff
@@ -3970,6 +3999,41 @@ void Editor::DoSearchReplace(const SearchReplaceDisposition Disposition)
 			{ lng::MOk });
 }
 
+void Editor::SaveFoundItemsToNewEditor(const VMenu& ListBox, const bool MatchingFilter)
+{
+	const auto ShellEditor{ FileEditor::create(GetSearchAllFileName(), GetCodePage(), FFILEEDIT_CANNEWFILE | FFILEEDIT_ENABLEF6) };
+	const auto NewEditor{ ShellEditor->GetEditor() };
+	const auto FilterFlags{ LIF_HIDDEN | (MatchingFilter ? LIF_FILTERED : 0) };
+
+	using namespace std::views;
+
+	for (const auto I : ListBox.GetItems()
+		| filter([FilterFlags](const auto& Item) { return !(Item.Flags & FilterFlags); })
+		| transform([](const auto& Item) { return std::any_cast<FindCoord>(Item.ComplexUserData).Line; })
+		| chunk_by(std::ranges::equal_to{})
+		| transform([](auto&& chunk) { return chunk.front(); }))
+	{
+		NewEditor->InsertString(GetStringByNumber(I)->GetString(), NewEditor->LastLine());
+		NewEditor->TextChanged(true);
```

Comment:
Do we need to call this N times?
Maybe once after the loop?

Reply:
It is inexpensive, but yes, it should be out of the loop.

Besides, in the version I am working on, I do not do it at all. Since, as we discussed above, I want to avoid asking to save the file if the user have not touched it, I do not mark it dirty. There are more things I am doing to achieve the behavior I want. I'll explain it separately.